community/detector/cves/CVE-2021-29441/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202129441/CVE202129441VulnDetector.java

```diff
@@ -0,0 +1,145 @@
+            + " any user to carry out any administrative tasks on the Nacos server.",
+    author = "hh-hunter",
+    bootstrapModule = CVE202129441DetectorBootstrapModule.class)
+public final class CVE202129441VulnDetector implements VulnDetector {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final String CHECK_VUL_PATH = "nacos/v1/auth/users/?pageNo=1&pageSize=9";
+  public static final String DETECTION_STRING = "pageItems";
+
+  private final HttpClient httpClient;
+  private final Clock utcClock;
+
+  @Inject
+  CVE202129441VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.httpClient = checkNotNull(httpClient);
+    this.utcClock = checkNotNull(utcClock);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("CVE-2921-29441 starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetUri =
+        NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + CHECK_VUL_PATH;
+    try {
+      HttpResponse httpResponse =
+          httpClient.send(
+              get(targetUri)
+                  .setHeaders(HttpHeaders.builder().addHeader(USER_AGENT, "Nacos-Server").build())
+                  .build(),
+              networkService);
+
+      if (httpResponse.status().code() == 200
+          && httpResponse.bodyString().get().contains(DETECTION_STRING)) {
+        return true;
+      }
+
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Request to target %s failed", networkService);
+      return false;
+    }
+    return false;
+  }
+
+  // This builds the DetectionReport message for a specific vulnerable network service.
+  private DetectionReport buildDetectionReport(
+      TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(
+                    VulnerabilityId.newBuilder()
+                        .setPublisher("TSUNAMI_COMMUNITY")
+                        .setValue("CVE_2021_29441"))
+                .setSeverity(Severity.CRITICAL)
+                .setTitle("CVE-2021-29441")
```

Comment:
Can you use a more descriptive title? For example, `CVE-2021-29441: Authentication Bypass in Nacos before 1.4.1`.